src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, ElementRef, EventEmitter, forwardRef, Input, Output} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+const NGB_CHECKBOX_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbCheckbox),
+  multi: true
+};
+
+let checkbox_id_generator = 1;
+
+/**
+ * A component that allows the user to edit boolean values.
+ */
+@Component({
+  selector: 'ngb-checkbox',
+  host: {'[class.custom-control]': 'true', '[class.custom-checkbox]': 'true'},
+  template: `
+    <input
+      type="checkbox"
+      class="custom-control-input"
+      [disabled]="disabled || readonly"
+      [checked]="value"
+      (change)="_valueChange($event)"
+      (blur)="_blur()"
+      [attr.id]="id"
+    />
+    <label [attr.tabindex]="-1" class="custom-control-label" [attr.for]="id">
+      <ng-content></ng-content>
+    </label>
+  `,
+  providers: [NGB_CHECKBOX_VALUE_ACCESSOR]
+})
+export class NgbCheckbox implements ControlValueAccessor {
+  private _onChange: (v: any) => void;
```

Comment:
this can be set as in NG inputs, e.g. `datepicker-input.ts`:
```ts
 private _onChange = (_: any) => {};
 private _onTouched = () => {};
```
this would remove the need of constructor implementation:
```ts
constructor(private _elementRef: ElementRef) {}
```